crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -132,6 +132,10 @@ pub struct Room {
     /// to disk but held in memory.
     #[cfg(all(feature = "e2e-encryption", feature = "experimental-sliding-sync"))]
     pub latest_encrypted_events: Arc<SyncRwLock<RingBuffer<Raw<AnySyncTimelineEvent>>>>,
+
+    /// The list of pinned event ids in this room in an observable format so
+    /// they can be subscribed to.
+    pub pinned_event_ids: SharedObservable<Vec<OwnedEventId>>,
```

Comment:
nit: not `pub`, since there's a getter for a subscriber

Also, would it make sense to use `ObservableVector` here? If we did that, and rejigger the API so that we only push new pinned event / remove pinned event, we could have a finer-grained API, but maybe it's not worth it?

Reply:
`ObservableVector` doesn't implement `Clone` so it can't be used in `Room`, which does.